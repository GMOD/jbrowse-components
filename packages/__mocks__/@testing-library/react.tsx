import React, { Suspense } from 'react'
import { createTheme, ThemeProvider } from '@mui/material/styles'

const react = jest.requireActual('@testing-library/react')
const render = (args: React.ReactNode) => {
  return react.render(
    <Suspense fallback={null}>
      <ThemeProvider theme={createTheme()}>{args}</ThemeProvider>
    </Suspense>,
  )
}

module.exports = { ...react, render }
