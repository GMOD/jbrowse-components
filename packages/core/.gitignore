.DS_Store
node_modules
coverage
dist
*.log
*.swp
*.js
*.d.ts
*.js.map
./BaseFeatureWidget
./ReExports
./TextSearch
./assemblyManager
./configuration
./data_adapterts
./pluggableElementTypes
./rpc
./ui
./util
