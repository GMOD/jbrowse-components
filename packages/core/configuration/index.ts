export {
  ConfigurationSchema,
  ConfigurationReference,
} from './configurationSchema'

export type {
  AnyConfigurationSchemaType,
  AnyConfigurationModel,
  AnyConfigurationSlot,
  AnyConfigurationSlotType,
  AnyConfiguration,
} from './types'

export * from './util'
