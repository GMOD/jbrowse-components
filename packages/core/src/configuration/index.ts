export {
  ConfigurationSchema,
  ConfigurationReference,
} from './configurationSchema'

export * from './util'
