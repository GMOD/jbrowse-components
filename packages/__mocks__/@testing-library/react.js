import { StylesProvider } from '@material-ui/core/styles'
import React from 'react'

const react = jest.requireActual('@testing-library/react')
const generateClassName = (rule, styleSheet) =>
  `${styleSheet.options.classNamePrefix}-${rule.key}`
const render = args => {
  return react.render(
    <StylesProvider generateClassName={generateClassName}>
      {args}
    </StylesProvider>,
  )
}

module.exports = { ...react, render }
