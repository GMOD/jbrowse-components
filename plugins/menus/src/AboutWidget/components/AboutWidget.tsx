import Typography from '@material-ui/core/Typography'
import { observer } from 'mobx-react'
import { IAnyStateTreeNode } from 'mobx-state-tree'
import { getSession } from '@jbrowse/core/util'
import React from 'react'
import { makeStyles } from '@material-ui/core'
import Link from '@material-ui/core/Link'

const useStyles = makeStyles(theme => ({
  root: {
    margin: theme.spacing(2),
  },
  subtitle: {
    margin: theme.spacing(),
  },
}))

function About({ model }: { model: IAnyStateTreeNode }) {
  const classes = useStyles()
  const root = model ? getSession(model) : { version: '' }
  return (
    <div className={classes.root}>
      <Typography variant="h4" align="center" color="primary">
        JBrowse 2
      </Typography>
      <Typography variant="h6" align="center" className={classes.subtitle}>
        {root.version}
      </Typography>
      <Typography align="center" variant="body2">
        JBrowse is a{' '}
        <Link href="http://gmod.org/" target="_blank" rel="noopener noreferrer">
          GMOD
        </Link>{' '}
        project
      </Typography>
      <br />
      <Typography align="center">
        Â© 2019 The Evolutionary Software Foundation
      </Typography>
      {/* <br />
      <Typography align="center">
        JBrowse is funded by the{' '}
        <Link
          href="https://genome.gov/"
          target="_blank"
          rel="noopener noreferrer"
        >
          NHGRI
        </Link>
      </Typography> */}
    </div>
  )
}

export default observer(About)
