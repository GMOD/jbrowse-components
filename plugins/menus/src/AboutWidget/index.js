import { ConfigurationSchema } from '@jbrowse/core/configuration'
import { ElementId } from '@jbrowse/core/util/types/mst'
import { types } from 'mobx-state-tree'

export const configSchema = ConfigurationSchema('AboutWidget', {})

export const stateModel = types.model('AboutWidget', {
  id: ElementId,
  type: types.literal('AboutWidget'),
})

export { default as ReactComponent } from './components/AboutWidget'
