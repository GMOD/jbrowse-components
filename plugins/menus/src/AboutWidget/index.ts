import { lazy } from 'react'

import { ConfigurationSchema } from '@jbrowse/core/configuration'
import { WidgetType } from '@jbrowse/core/pluggableElementTypes'
import { ElementId } from '@jbrowse/core/util/types/mst'
import { types } from 'mobx-state-tree'

import type PluginManager from '@jbrowse/core/PluginManager'


const configSchema = ConfigurationSchema('AboutWidget', {})

const stateModel = types.model('AboutWidget', {
  id: ElementId,
  type: types.literal('AboutWidget'),
})

export default function AboutWidgetF(pluginManager: PluginManager) {
  pluginManager.addWidgetType(() => {
    return new WidgetType({
      name: 'AboutWidget',
      heading: 'About',
      configSchema,
      stateModel,
      ReactComponent: lazy(() => import('./components/AboutWidget')),
    })
  })
}
