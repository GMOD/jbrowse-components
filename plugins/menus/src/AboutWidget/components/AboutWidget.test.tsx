import React from 'react'
import { ConfigurationSchema } from '@jbrowse/core/configuration'
import { render } from '@testing-library/react'
import { types } from 'mobx-state-tree'

// locals
import AboutWidget from './AboutWidget'

describe('<AboutWidget />', () => {
  it('renders', () => {
    const session = types
      .model({
        configuration: ConfigurationSchema('Null', {}),
        rpcManager: types.frozen({}),
        widgetModel: types.model({ type: types.literal('AboutWidget') }),
      })
      .create(
        { widgetModel: { type: 'AboutWidget' } },
        {
          pluginManager: {
            pluginMetadata: {},
            plugins: [
              {
                name: 'HelloPlugin',
                version: '1.0.0',
                url: 'http://google.com',
              },
            ],
          },
        },
      )
    const { container } = render(<AboutWidget model={session.widgetModel} />)
    expect(container).toMatchSnapshot()
  })
})
