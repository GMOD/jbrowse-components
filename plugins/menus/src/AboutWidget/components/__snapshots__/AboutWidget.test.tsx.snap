// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AboutWidget /> renders 1`] = `
<div>
  <div
    class="css-1a8qml5-root"
  >
    <h4
      class="MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter css-1ya2hdm-MuiTypography-root"
      style="--Typography-textAlign: center;"
    >
      JBrowse 2
    </h4>
    <h6
      class="MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter css-6je6nr-MuiTypography-root-subtitle"
      style="--Typography-textAlign: center;"
    />
    <p
      class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-1j3qii7-MuiTypography-root"
      style="--Typography-textAlign: center;"
    >
      JBrowse is a
       
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1orsvah-MuiTypography-root-MuiLink-root"
        href="http://gmod.org/"
        rel="noopener noreferrer"
        target="_blank"
      >
        GMOD
      </a>
       
      project
    </p>
    <br />
    <p
      class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-1j3qii7-MuiTypography-root"
      style="--Typography-textAlign: center;"
    >
      Â© 2019-2022 The Evolutionary Software Foundation
    </p>
    <div
      class="css-la8yjp-pluginList"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
      >
        External plugins loaded
      </p>
      <ul>
        <li>
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1orsvah-MuiTypography-root-MuiLink-root"
            href="http://google.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            HelloPlugin 1.0.0
          </a>
        </li>
      </ul>
      <p
        class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
      >
        Core plugins loaded
      </p>
      <ul />
    </div>
  </div>
</div>
`;
